server/portal/apps/intromessages/views.py

```diff
@@ -0,0 +1,43 @@
+"""
+.. :module: apps.intromessages.views
+   :synopsis: Views to handle read/unread status of IntroMessages
+"""
+
+import logging
+from portal.views.base import BaseApiView
+from django.http import JsonResponse
+from portal.apps.intromessages.models import IntroMessages
+from django.contrib.auth.decorators import login_required
+from django.utils.decorators import method_decorator
+import json
+
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(login_required, name='dispatch')
+class IntroMessagesView(BaseApiView):
+    # Get all of the IntroMessages that have been read (unread = False)
+    def get(self, request, *args, **kwargs):
+        messages_array = IntroMessages.objects.filter(user=request.user).values('component', 'unread').values()
```

Comment:
If you remove the trailing `.values()` call, you end up with a QuerySet of just the desired `component` and `unread` values, so you wouldn't have to filter again with list comprehension below
```suggestion
        messages_array = IntroMessages.objects.filter(user=request.user).values('component', 'unread')
```